hbase-handler/src/java/org/apache/hadoop/hive/hbase/HBaseStorageHandler.java

```diff
@@ -296,7 +297,8 @@ public URI getURIForAuth(Table table) throws URISyntaxException {
     String hbase_host = tableProperties.containsKey(HBASE_HOST_NAME)? tableProperties.get(HBASE_HOST_NAME) : hbaseConf.get(HBASE_HOST_NAME);
     String hbase_port = tableProperties.containsKey(HBASE_CLIENT_PORT)? tableProperties.get(HBASE_CLIENT_PORT) : hbaseConf.get(HBASE_CLIENT_PORT);
     String table_name = tableProperties.getOrDefault(HBaseSerDe.HBASE_TABLE_NAME, null);
-    String column_family = tableProperties.getOrDefault(HBaseSerDe.HBASE_COLUMNS_MAPPING, null);
+    String column_family =
+        URLEncoder.encode(tableProperties.getOrDefault(HBaseSerDe.HBASE_COLUMNS_MAPPING, null));
     if (column_family != null)
       return new URI(HBASE_PREFIX+"//"+hbase_host+":"+hbase_port+"/"+table_name+"/"+column_family);
```

Comment:
Why don't we encode the whole string before passing to the URI constructor?

Reply:
Oh, I just wrote the whole comment above addressing this before I saw yours.  Please read what I wrote above.

Perhaps it is better to encode the whole string?  I just was worried about backward compatibility and the potential for URL Encoding characters that might get by in the URI...For instance, we found that one "#" characters worked fine in Serde properties, but we only hit the problem when there were 2 '#' characters.  I'm a bit worried about breaking something.

Reply:
Changed the code so we re resolving the whole URI now